controllers/actions.github.com/autoscalingrunnerset_controller.go

```diff
@@ -484,6 +656,13 @@ func (r *AutoscalingRunnerSetReconciler) deleteRunnerScaleSet(ctx context.Contex
 		return err
 	}
 
+	err = patch(ctx, r.Client, autoscalingRunnerSet, func(obj *v1alpha1.AutoscalingRunnerSet) {
```

Comment:
Had to clean this up now that we clear permissions. We get client from secret, and we keep reconciling which can return an error trying to fetch the client.
So once the scale set is deleted, we remove the annotation signalling this does not have to be done any more